### PR TITLE
Migrate 'LegendRamp' component from makeStyles to styled-component + cleanup

### PR DESCRIPTION
**Pull request by jantolg**
# Description

This PR includes a refactorization of LegendRamp component, all classes from makestyles are been converted to styled components

Shortcut:  [297037](https://app.shortcut.com/cartoteam/story/297037/migrate-away-from-makestyles-i-widgets)

## Type of change

- Refactor

# Acceptance

The code resultant must return same styles component visualization as the same before changes

1. Overview a general visualization from v1.x to v2.x

# Basic checklist

- Good PR name
- Shortcut link
- Changelog entry
- Just one issue per PR
- GitHub labels
- Proper status & reviewers
- Tests
- Documentation

### CODE REVIEW

**Comment by vmilan** [comment]
> A couple of test are failing due to weird behaviour of getComputedStyles related with background and linear-gradient, this function doesn't return background value when linear-gradient are applied. I need research a bit more

ok, let us know when this PR is ready for review please. Thanks!!

**Comment by aaranadev** [comment]
@vmilan the PR is ready to check it out.

File: packages/react-ui/src/widgets/legend/LegendRamp.js [hunk 1]
```diff
@@ -52,7 +43,7 @@ function LegendRamp({ isContinuous = false, legend }) {
         <>
           <Grid container item>
             {isContinuous ? (
-              <StepsContinuous palette={palette} />
+              <StepsContinuous className="step" item xs palette={palette} />
```
Comment (VictorVelarde) [inline packages/react-ui/src/widgets/legend/LegendRamp.js:46]: Are we sure about this className?

File: packages/react-ui/src/widgets/legend/LegendRamp.js [hunk 2]
```diff
@@ -95,36 +86,28 @@ LegendRamp.propTypes = {
...
-const useStylesStepsContinuous = makeStyles(() => ({
-  step: {
-    height: 8,
-    borderRadius: 4
-  }
+const StepsContinuous = styled(Grid, {
+  shouldForwardProp: (prop) => prop !== 'palette'
+})(({ palette }) => ({
+  height: 8,
+  borderRadius: 4,
```
Comment (vmilan) [inline packages/react-ui/src/widgets/legend/LegendRamp.js:93]: replace values by theme.spacing() please

File: packages/react-ui/src/widgets/legend/LegendRamp.js [hunk 3]
```diff
@@ -95,36 +86,28 @@ LegendRamp.propTypes = {
...
+const StepGrid = styled(Grid, {
+  shouldForwardProp: (prop) => prop !== 'color'
+})(({ color }) => ({
+  height: 8,
+  backgroundColor: color,
+  '&:first-of-type': {
+    borderRadius: '4px 0 0 4px'
+  },
+  '&:last-of-type': {
+    borderRadius: '0 4px 4px 0'
```
Comment (vmilan) [inline packages/react-ui/src/widgets/legend/LegendRamp.js:106]: replace values by theme.spacing() please

Comment (jantolg) [reply packages/react-ui/src/widgets/legend/LegendRamp.js:106]: done!

File: packages/react-ui/src/widgets/legend/LegendRamp.js (hunk 2 above)
Comment (jantolg) [reply packages/react-ui/src/widgets/legend/LegendRamp.js:93]: done!

File: packages/react-ui/src/widgets/legend/LegendRamp.js (hunk 1 above)
Comment (jantolg) [reply packages/react-ui/src/widgets/legend/LegendRamp.js]: done!

**Review by vmilan — APPROVED** [review]
LGTM

**Pull request merged**